packages/teleport/src/Discover/LoginTrait/LoginTrait.tsx

```diff
@@ -0,0 +1,169 @@
+  const [newLogin, setNewLogin] = useState('');
+  const [showInputBox, setShowInputBox] = useState(false);
+
+  return (
+    <Box>
+      <Header>Set Up Access</Header>
+      {attempt.status === 'processing' && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {attempt.status === 'failed' && <Danger>{attempt.statusText}</Danger>}
+      {attempt.status === 'success' && (
+        <>
+          <Text bold mb={2}>
+            Select or Add Linux Principles
+          </Text>
+          <Box mb={6}>
+            {Object.keys(loginMap).map((login, index) => {
+              return (
+                <Flex
+                  key={index}
+                  p={2}
+                  mb={1}
+                  borderRadius={2}
+                  width="300px"
+                  alignItems="center"
+                  borderColor="colors.primary.light"
+                  css={`
+                    border: 1px solid
+                      ${props => props.theme.colors.primary.light};
+                  `}
+                >
+                  <input
+                    type="checkbox"
+                    name={login}
+                    checked={loginMap[login]}
+                    css={`
+                      margin-right: 10px;
+                      accent-color: ${props =>
+                        props.theme.colors.secondary.main};
+                      &:hover {
+                        cursor: pointer;
+                      }
+                    `}
+                    onChange={() => toggleLoginSelect(login)}
+                  />
+                  <label
+                    htmlFor={login}
+                    css={`
+                      width: 250px;
+                      overflow: hidden;
+                      text-overflow: ellipsis;
+                    `}
+                  >
+                    {login}
+                  </label>
+                </Flex>
+              );
+            })}
+            {showInputBox && (
+              <Flex alignItems="end" mt={3}>
+                <Validation>
+                  <FieldInput
+                    placeholder="name"
+                    autoFocus
+                    width="200px"
+                    value={newLogin}
+                    type="text"
+                    onChange={e => setNewLogin(e.target.value.trim())}
+                    mr={3}
+                    mb={0}
+                  />
+                  <ButtonPrimary
+                    size="small"
+                    mb={2}
+                    disabled={newLogin.length === 0}
+                    onClick={() => {
+                      addLogin(newLogin);
+                      setNewLogin('');
+                      setShowInputBox(false);
+                    }}
+                  >
+                    Add
+                  </ButtonPrimary>
+                </Validation>
+              </Flex>
+            )}
+            {!showInputBox && (
```

Comment:
This could be `||` instead as we're checking `showInputBox` above

Reply:
hm, do you mean `{showInputBox || ...`? If yes, it looks a bit confusing to me, and will keep `&&`

Reply:
You can chain onto the `{showInputBox && (` above by either `{showInputBox && (<element>) || (<other>)}` or `{showInputBox ? (<element>) : (<other>)}`